torchtune/utils/_checkpointing/_checkpointer_utils.py

```diff
@@ -54,12 +57,19 @@ def safe_torch_load(checkpoint_path: Path, weights_only: bool = True) -> Dict[st
     try:
         # convert the path into a string since pathlib Path and mmap don't work
         # well together
-        state_dict = torch.load(
-            str(checkpoint_path),
-            map_location="cpu",
-            mmap=True,
-            weights_only=weights_only,
+        is_safetensors_file = (
+            True if str(checkpoint_path).endswith(".safetensors") else False
         )
+        if is_safetensors_file:
+            result = {}
+            with safe_open(checkpoint_path, framework="pt", device="cpu") as f:
+                for k in f.keys():
+                    result[k] = f.get_tensor(k)
+            state_dict = result
+        else:
+            state_dict = torch.load(
+                str(checkpoint_path), map_location="cpu", mmap=True, weights_only=True
```

Comment:
looks like `weights_only` arg is not passed around here?

Reply:
Oh yes, but I just looked at the latest version and it has been updated.
```
state_dict = torch.load(
                str(checkpoint_path),
                map_location="cpu",
                mmap=True,
                weights_only=weights_only,
            )
```